app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java

```diff
@@ -211,8 +214,49 @@ else if (mCafeData instanceof DiningHallModel) {
             }
         }
 
-        // Set up wait times feature.
-        mWaitTimesFragment = new WaitTimesFragment();
+        this.setupWaitTimes();
+    }
+
+    // Set up wait times feature.
+    private void setupWaitTimes() {
+        // Return if eatery is not a campus eatery / no swipe data available.
+        if (!(mCafeData instanceof CampusModel) || ((CampusModel) mCafeData).getSwipeData().size() == 0) {
+            return;
+        }
+
+        // Data for wait times chart.
+        List<Swipe> swipeData = new ArrayList<Swipe>();
+
+        for (Swipe s: ((CampusModel) mCafeData).getSwipeData()) {
+            Log.i("swipeDataLog:", "\n" + "\n"
+                    + "\n Start: " + s.getStart().getHour() + ", End: " + s.getEnd().getHour()
+                    + "\n Density: " + s.swipeDensity
+                    + "\n Low: " + s.waitTimeLow + ", End: " + s.waitTimeHigh
+            );
+        }
+
+        // Wait times chart must have 21 elements for each hour within 6am - 3am.
+        // i represents the hour, with 0 being 6am - 7am.
+        for (int i = 0; i < 21; i++) {
+            // Default swipeDensity, waitTimeLow, waitTimeHigh = 0 for hours without backend swipe data.
+            double swipeDensity = 0;
+            int waitTimeLow = 0, waitTimeHigh = 0;
+            // Parse through backend swipe data. If multiple swipe data found for the hour, use maximum.
+            for (Swipe s: ((CampusModel) mCafeData).getSwipeData()) {
+                int hourFromSwipe = s.getStart().getHour();
+                hourFromSwipe = hourFromSwipe < 6 ? (18 + hourFromSwipe) : hourFromSwipe - 6;
+                if (hourFromSwipe == i) {
+                    swipeDensity = Math.max(swipeDensity, s.swipeDensity);
+                    waitTimeLow = Math.max(waitTimeLow, s.waitTimeLow);
+                    waitTimeHigh = Math.max(waitTimeHigh, s.waitTimeHigh);
+                }
+            }
+            // Index of swipeData now represents start/end with 0 being start = 6am, end = 7am.
+            swipeData.add(new Swipe(null, null, swipeDensity, waitTimeLow, waitTimeHigh));
```

Comment:
what's the purpose of the `start` and `end` field in the `Swipe` object if it's always null

Reply:
i read your note again. if you don't believe that we need `LocalTime` then let's remove these two params from `Swipe` instead of setting them to null

Reply:
We need LocalTime start/end when we get the hours from backend, so we can't remove them from Swipe.
Backend only has data for certain hours, whereas we need to display wait times for all hours 6am - 3am. 
I didn't think it'd be necessary to create LocalTime start/end for the hours which had no data from backend since we'd be doing .getStart().getHour() again to get the hour from LocalTime, so I just set them to null and used indexing of the arraylist instead.

Reply:
tbh i'm still a bit confused. the start/end param in `Swipe` is annotated with `@NonNull` how did we even bypass the annotation? 

also, if we don't use start/end time from the backend, then why don't we remove the call for start/end from the graphql query?

Reply:
Just updated this. Our chart component needed a list of 21 swipe data objects representing wait-times, yet the backend would only return a list of hours which it had data for. I changed it up and built a separate constructor for creating a blank swipe object when we have no data, and just adding the swipe object to the list when we do. Hopefully this clears things up here

Reply:
Oh also, we do still need the start/end data for the swipe (need this when we decide where to place the swipe in the ordered array), but I simplified the model in that we don't need to extend Comparator anymore